benchmarks/graphics_pipeline/main.cpp

```diff
@@ -30,6 +31,37 @@
 
 using namespace ppx;
 
+static constexpr float kCameraSpeed = 0.2f;
+
+class FreeCamera
+    : public ppx::PerspCamera
+{
+public:
+    enum class MovementDirection
+    {
+        FORWARD,
+        LEFT,
+        RIGHT,
+        BACKWARD
+    };
+
+    FreeCamera()
+    {
+        Setup();
+    }
+
+    // Moves the location of the camera in dir direction for distance units.
+    void Move(MovementDirection dir, float distance);
+
+private:
+    // Initializes the eye position and target of the camera.
+    void Setup()
+    {
+        mEyePosition = float3(0, 0, -5);
+        mTarget      = float3(0, 0, -4);
```

Comment:
this should move in the constructor. Setup is called once.

Reply:
Done.